### PR TITLE
JD- settings implementation

### PR DESCRIPTION
**Pull request by VickyRavi**

### CODE REVIEW

File: src/main/java/com/sellinall/jdlisting/Main.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.sellinall.jdlisting;
```
Comment (Deepakelango) [inline src/main/java/com/sellinall/jdlisting/Main.java]: Reviewed

File: src/main/resources/CamelContext.xml [hunk 2]
```diff
@@ -39,7 +83,83 @@
 
 	<!-- The camel route context -->
 	<camelContext id="camel" xmlns="http://camel.apache.org/schema/spring">
+		<route id="addAccount">
+			<from uri="direct:addAccount" />
+			<process ref="initializeAccountRoute" />
+			<when>
+				<simple>${exchangeProperty.failureReason} == null</simple>
```
Comment (ramchnk) [inline src/main/resources/CamelContext.xml]: follow template server

Comment (ramchnk) [reply src/main/resources/CamelContext.xml]: image

File: src/main/resources/CamelContext.xml [hunk 3]
```diff
@@ -39,7 +83,83 @@
...
+			<when>
+				<simple>${exchangeProperty.needToValidateAccount} == true</simple>
+				<to uri="direct:submitJDAuthentication" />
+			</when>
+			<process ref="updateAccountDetails" />
+			<process ref="constructResponseMessage" />
+		</route>
+
+		<route id="submitJDAuthentication">
+			<from uri="direct:submitJDAuthentication" />
```
Comment (ramchnk) [inline src/main/resources/CamelContext.xml]: change route name as submitAuthentication

File: src/main/java/com/sellinall/jdlisting/db/AccountDBQuery.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		exchange.setProperty("accountDetails", outBody);
+	}
+
+	private DBObject createBody(Exchange exchange) {
+		String accountNumber = (String) exchange.getProperty("accountNumber");
+		MongoCollection<Document> table = DbUtilities.getDBCollection("accounts");
+		BasicDBObject searchQuery = new BasicDBObject();
+		ObjectId objid = new ObjectId(accountNumber);
+		searchQuery.put("_id", objid);
+		Document accountDocument = table.find(searchQuery).first();
```
Comment (ramchnk) [inline src/main/java/com/sellinall/jdlisting/db/AccountDBQuery.java]: table.findOne()

File: src/main/java/com/sellinall/jdlisting/db/LoadAccountDataByNicknameId.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+		exchange.setProperty("clientSecret", postHelper.getString("clientSecret"));
+		exchange.setProperty("accessToken", EncryptionUtil.decrypt(postHelper.getString("accessToken")));
+		exchange.setProperty("countryCode", jd.getString("countryCode"));
+	}
+
+	private DBObject getAccountDetails(Exchange exchange) {
+		String accountNumber = (String) exchange.getProperty("accountNumber");
+		MongoCollection<Document> table = DbUtilities.getDBCollection("accounts");
+		String nickNameId = exchange.getProperty("nickNameID", String.class);
+		String jd = nickNameId.split("-")[0];
```
Comment (ramchnk) [inline src/main/java/com/sellinall/jdlisting/db/LoadAccountDataByNicknameId.java]: no need this one directly can apply

File: src/main/java/com/sellinall/jdlisting/Main.java (hunk 1 above)
Comment (ramchnk) [reply src/main/java/com/sellinall/jdlisting/Main.java]: changes checked

**Pull request merged**